
<!DOCTYPE html>
<html>
  <head>
    <title>Communities API API</title>
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <script type="text/javascript" src="bootstrap.min.js"></script>
    <link rel="stylesheet" type="text/css" href="bootstrap.min.css" />
  </head>

  <body style="background-color: white; margin: 15px;">
    <div class="navbar navbar-inverse navbar-fixed-top">
      <div class="navbar-inner">
	<div class="container">
	  <a class="brand" href="./index.html">Communities API API documentation</a>
	</div>
      </div>
    </div>

    <div class="row" style="margin-top: 50px;">
      <div name="index" class="span3" style="position: fixed; width: 130px;">
	<ul class="nav nav-list">
	  <li class="nav-header">Resources</li>
<li><a href="#overview">overview</a></li><li><a href="#annotation">annotation</a></li><li><a href="#compute">compute</a></li><li><a href="#download">download</a></li><li><a href="#heartbeat">heartbeat</a></li><li><a href="#inbox">inbox</a></li><li><a href="#library">library</a></li><li><a href="#m5nr">m5nr</a></li><li><a href="#matrix">matrix</a></li><li><a href="#metadata">metadata</a></li><li><a href="#metagenome">metagenome</a></li><li><a href="#profile">profile</a></li><li><a href="#project">project</a></li><li><a href="#sample">sample</a></li><li><a href="#server">server</a></li><li><a href="#submission">submission</a></li><li><a href="#validation">validation</a></li></ul>	
    </div>
    <div class="span12" style="margin-left: 270px;">
<h1><a name='overview' style='padding-top: 50px;'>Communities API Overview</a></h1><p><p>The MG-RAST API covers most of the functionality available through the MG-RAST website, with access to annotations, analyses, metadata and access to the MG-RAST user inbox to view contents as well as upload files. All sequence data and data products from intermediate stages in the analysis pipeline are available for download. Other resources provide services not available through the website, e.g. the m5nr resource lets you query the m5nr database.</p><p>Each query to the API is represented as a URI beginning with "http://kbase.us/services/communities/" and has a defined structure to pass the requests and parameters to the API server.</p><p>The URI queries can be used from the command line, e.g. using curl, in a browser, or incorporated in a shell script or program.</p><p>Each URI has the form</p><pre>"http://kbase.us/services/communities/{version}/{resourcepath}?{querystring}"</pre><p>The {version} value (currently '1') explicitly directs the request to a specific version of the API, if it is omitted the latest API version will be used.</p><p>The resource path is constructed from the path parameters listed below to define a specific resource and the optional query string is used to filter the results obtained for the resource. For example:</p><pre>http://kbase.us/services/communities/1/annotation/sequence/mgm4447943.3?evalue=10&type=organism&source=SwissProt</pre><p>In this example the resource path "annotation/sequence/mgm4447943.3" defines a request for the annotated sequences for the MG-RAST job with ID 4447943.3.</p><p>The optional query string "evalue=10&type=organism&source=SwissProt" modifies the results by setting an evalue cutoff, annotation type and database source.</p><p>The API provides an authentication mechanism for access to private MG-RAST jobs and users' inbox. The 'auth_key' (or 'webkey') is a 25 character long string  (e.g. 'j6FNL61ekNarTgqupMma6eMx5') which is used by the API to identify an MG-RAST user account and determine access rights to metagenomes. Note that the auth_key is valid for a limited time after which queries using the key will be rejected. You can create a new auth_key or view the expiration date and time of an existing auth_key on the MG-RAST website. An account can have only one valid auth_key and creating a new key will invalidate an existing key.</p><p>All public data in MG-RAST is available without an auth_key. All API queries for private data which either do not have an auth_key or use an invalid or expired auth_key will get a "insufficient permissions to view this data" response.</p><p>The auth_key can be included in the query string like:</p><pre>
http://kbase.us/services/communities/1/annotation/sequence/mgm4447943.3?evalue=10&type=organism&source=SwissProt&auth=j6FNL61ekNarTgqupMma6eMx5</pre><p>or in a request using curl like:</p><pre>curl -X GET -H "auth: j6FNL61ekNarTgqupMma6eMx5" "http://kbase.us/services/communities/1/annotation/sequence/mgm4447943.3?evalue=10&type=organism&source=SwissProt"</pre><p>Note that for the curl command the quotes are necessary for the query to be passed to the API correctly.</p><p>If an optional parameter passed through the query string has a list of values only the first will be used. When multiple values are required, e.g. for multiple md5 checksum values, they can be passed to the API like:</p><pre>curl -X POST -d '{"data":["000821a2e2f63df1a3873e4b280002a8","15bf1950bd9867099e72ea6516e3d602"]}' "http://kbase.us/services/communities/m5nr/md5"</pre><p>In some cases, the data requested is in the form of a list with a large number of entries. In these cases the limit and offset parameters can be used to step through the list, for example:</p><pre>http://kbase.us/services/communities/1/project?order=name&limit=20&offset=100</pre><p>will limit the number of entries returned to 20 with an offset of 100. If these parameters are not provided default values of limit=10 and offset=0 are used. The returned JSON structure will contain the 'next' and 'prev' (previous) URIs to simplify stepping through the list.</p><p>The data returned may be plain text, compressed gzipped files or a JSON structure.</p><p>Most API queries are 'synchronous' and results are returned immediately. Some queries may require a substantial time to compute results, in these cases you can select the asynchronous option by adding '&asynchronous=1' to the end of the query string. This query will then return a URL which will return the query results when they are ready.</p></p><hr><h1><a name='annotation' style='padding-top: 50px;'>annotation</a></h1><h2>Description</h2><p>All annotations of a metagenome for a specific annotation type and source</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>sequence - request</h2><h3>Description</h3><p>tab delimited annotated sequence stream</p><p>This is a stream GET request.</p><h3>Parameters</h3><ul><li><b>evalue</b> (int)</p><p>negative exponent value for maximum e-value cutoff: default is 5</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter</b> (string)</p><p>text string to filter annotations by: only return those that contain text</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_level</b> (string)</p><p>hierarchal level to filter annotations by, for organism or ontology only</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>identity</b> (int)</p><p>percent value for minimum % identity cutoff: default is 60</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>length</b> (int)</p><p>value for minimum alignment length cutoff: default is 15</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>RefSeq</b> - protein database, type organism, function, feature</li><li><b>GenBank</b> - protein database, type organism, function, feature</li><li><b>IMG</b> - protein database, type organism, function, feature</li><li><b>SEED</b> - protein database, type organism, function, feature</li><li><b>TrEMBL</b> - protein database, type organism, function, feature</li><li><b>SwissProt</b> - protein database, type organism, function, feature</li><li><b>PATRIC</b> - protein database, type organism, function, feature</li><li><b>KEGG</b> - protein database, type organism, function, feature</li><li><b>RDP</b> - RNA database, type organism, function, feature</li><li><b>Greengenes</b> - RNA database, type organism, function, feature</li><li><b>LSU</b> - RNA database, type organism, function, feature</li><li><b>SSU</b> - RNA database, type organism, function, feature</li><li><b>Subsystems</b> - ontology database, type ontology only</li><li><b>NOG</b> - ontology database, type ontology only</li><li><b>COG</b> - ontology database, type ontology only</li><li><b>KO</b> - ontology database, type ontology only</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>organism</b> - return organism data</li><li><b>function</b> - return function data</li><li><b>ontology</b> - return ontology data</li><li><b>feature</b> - return feature data</li><li><b>md5</b> - return md5sum data</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique metagenome identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/annotation/sequence/mgm4447943.3?evalue=10&type=organism&source=SwissProt<li>all annotated read sequences from mgm4447943.3 with hits in SwissProt organisms at evalue < e-10</ul><h3>Return Attributes</h3><ul><li><b>streaming text</b> (object)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>col_01 (string)</b> - sequence id</li><li><b>col_02 (string)</b> - m5nr id (md5sum)</li><li><b>col_03 (string)</b> - dna sequence</li><li><b>col_04 (string)</b> - semicolon separated list of annotations</li></ul><br></p></ul><h2>similarity - request</h2><h3>Description</h3><p>tab delimited blast m8 with annotation</p><p>This is a stream GET request.</p><h3>Parameters</h3><ul><li><b>evalue</b> (int)</p><p>negative exponent value for maximum e-value cutoff: default is 5</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter</b> (string)</p><p>text string to filter annotations by: only return those that contain text</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_level</b> (string)</p><p>hierarchal level to filter annotations by, for organism or ontology only</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>identity</b> (int)</p><p>percent value for minimum % identity cutoff: default is 60</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>length</b> (int)</p><p>value for minimum alignment length cutoff: default is 15</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>RefSeq</b> - protein database, type organism, function, feature</li><li><b>GenBank</b> - protein database, type organism, function, feature</li><li><b>IMG</b> - protein database, type organism, function, feature</li><li><b>SEED</b> - protein database, type organism, function, feature</li><li><b>TrEMBL</b> - protein database, type organism, function, feature</li><li><b>SwissProt</b> - protein database, type organism, function, feature</li><li><b>PATRIC</b> - protein database, type organism, function, feature</li><li><b>KEGG</b> - protein database, type organism, function, feature</li><li><b>RDP</b> - RNA database, type organism, function, feature</li><li><b>Greengenes</b> - RNA database, type organism, function, feature</li><li><b>LSU</b> - RNA database, type organism, function, feature</li><li><b>SSU</b> - RNA database, type organism, function, feature</li><li><b>Subsystems</b> - ontology database, type ontology only</li><li><b>NOG</b> - ontology database, type ontology only</li><li><b>COG</b> - ontology database, type ontology only</li><li><b>KO</b> - ontology database, type ontology only</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>organism</b> - return organism data</li><li><b>function</b> - return function data</li><li><b>ontology</b> - return ontology data</li><li><b>feature</b> - return feature data</li><li><b>md5</b> - return md5sum data</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique metagenome identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/annotation/similarity/mgm4447943.3?identity=80&type=function&source=KO<li>all annotated read blat stats from mgm4447943.3 with hits in KO functions at % identity > 80</ul><h3>Return Attributes</h3><ul><li><b>streaming text</b> (object)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>col_01 (string)</b> - query sequence id</li><li><b>col_02 (string)</b> - hit m5nr id (md5sum)</li><li><b>col_03 (float)</b> - percentage identity</li><li><b>col_04 (int)</b> - alignment length,</li><li><b>col_05 (int)</b> - number of mismatches</li><li><b>col_06 (int)</b> - number of gap openings</li><li><b>col_07 (int)</b> - query start</li><li><b>col_08 (int)</b> - query end</li><li><b>col_09 (int)</b> - hit start</li><li><b>col_10 (int)</b> - hit end</li><li><b>col_11 (float)</b> - e-value</li><li><b>col_12 (float)</b> - bit score</li><li><b>col_13 (string)</b> - semicolon separated list of annotations</li></ul><br></p></ul><hr><h1><a name='compute' style='padding-top: 50px;'>compute</a></h1><h2>Description</h2><p>Calculate various statistics for given input data.</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>alphadiversity - request</h2><h3>Description</h3><p>Calculate alpha diversity value for given ID and taxon level.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>strain</b> - bottom organism taxonomic level</li><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>RefSeq</b> - protein database, type organism, function, feature</li><li><b>GenBank</b> - protein database, type organism, function, feature</li><li><b>IMG</b> - protein database, type organism, function, feature</li><li><b>SEED</b> - protein database, type organism, function, feature</li><li><b>TrEMBL</b> - protein database, type organism, function, feature</li><li><b>SwissProt</b> - protein database, type organism, function, feature</li><li><b>PATRIC</b> - protein database, type organism, function, feature</li><li><b>KEGG</b> - protein database, type organism, function, feature</li><li><b>RDP</b> - RNA database, type organism, function, feature</li><li><b>Greengenes</b> - RNA database, type organism, function, feature</li><li><b>LSU</b> - RNA database, type organism, function, feature</li><li><b>SSU</b> - RNA database, type organism, function, feature</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique object identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/compute/alphadiversity/mgm4447943.3?level=order<li>retrieve alpha diversity for order taxon</ul><h3>Return Attributes</h3><ul><li><b>data</b> (float)</li><p>alpha diversity value</p><li><b>id</b> (string)</li><p>unique metagenome identifier</p><li><b>url</b> (string)</li><p>resource location of this object instance</p></p></ul><h2>normalize - request</h2><h3>Description</h3><p>Calculate normalized values for given input data.</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>columns</b> (list of string)</li><p>column id</p><p>This parameter must be passed in the message body.</p><li><b>data</b> (list of list)</li><p>ARRAY(0x1e2ae60)</p><p>This parameter must be passed in the message body.</p><li><b>norm</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>DESeq_blind</b> - DESeq_blind normalization method</li><li><b>standardize</b> - standardize normalization method</li><li><b>quantile</b> - quantile normalization method</li><li><b>DESeq_per_condition</b> - DESeq_per_condition normalization method</li><li><b>DESeq_pooled</b> - DESeq_pooled normalization method</li><li><b>DESeq_pooled_CR</b> - DESeq_pooled_CR normalization method</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>rows</b> (list of string)</li><p>row id</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"columns": ["mgm4441619.3","mgm4441656.4","mgm4441680.3","mgm4441681.3"], "rows": ["Eukaryota","Bacteria","Archaea"], "data": [[135,410,848,1243],[4397,6529,71423,204413],[1422,2156,874,1138]]}' "http://kbase.us/services/communities/1/compute/normalize"<li>retrieve normalized values for input abundances</ul><h3>Return Attributes</h3><ul><li><b>columns</b> (list of strings)</li><p>column id</p><li><b>data</b> (list of lists of floats)</li><p>normalized value</p><li><b>rows</b> (list of strings)</li><p>row id</p></p></ul><h2>significance - request</h2><h3>Description</h3><p>Calculate significance values for given input data.</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>columns</b> (list of string)</li><p>column id</p><p>This parameter must be passed in the message body.</p><li><b>data</b> (list of list)</li><p>ARRAY(0x20e2d50)</p><p>This parameter must be passed in the message body.</p><li><b>groups</b> (list of string)</li><p>group name</p><p>This parameter must be passed in the message body.</p><li><b>norm</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>DESeq_blind</b> - DESeq_blind normalization method</li><li><b>standardize</b> - standardize normalization method</li><li><b>quantile</b> - quantile normalization method</li><li><b>DESeq_per_condition</b> - DESeq_per_condition normalization method</li><li><b>DESeq_pooled</b> - DESeq_pooled normalization method</li><li><b>DESeq_pooled_CR</b> - DESeq_pooled_CR normalization method</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>raw</b> (boolean)</p><p>option to use raw data (not normalize)</p><p>This parameter must be passed in the message body.</p><li><b>rows</b> (list of string)</li><p>row id</p><p>This parameter must be passed in the message body.</p><li><b>test</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>Kruskal-Wallis</b> - Kruskal-Wallis significance testing method</li><li><b>t-test-paired</b> - t-test-paired significance testing method</li><li><b>Wilcoxon-paired</b> - Wilcoxon-paired significance testing method</li><li><b>t-test-unpaired</b> - t-test-unpaired significance testing method</li><li><b>Mann-Whitney-unpaired-Wilcoxon</b> - Mann-Whitney-unpaired-Wilcoxon significance testing method</li><li><b>ANOVA-one-way</b> - ANOVA-one-way significance testing method</li></ul><br><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"test":"Kruskal-Wallis","groups":["whale","whale","cow","cow"],"columns": ["mgm4441619.3","mgm4441656.4","mgm4441680.3","mgm4441681.3"], "rows": ["Eukaryota","Bacteria","Archaea"], "data": [[135,410,848,1243],[4397,6529,71423,204413],[1422,2156,874,1138]]}' "http://kbase.us/services/communities/1/compute/significance"<li>retrieve significance values for input abundances and groups using the 'Kruskal-Wallis' significance test</ul><h3>Return Attributes</h3><ul><li><b>columns</b> (list of strings)</li><p>column name</p><li><b>data</b> (list of lists of floats)</li><p>significance value</p><li><b>rows</b> (list of strings)</li><p>row name</p></p></ul><h2>heatmap - request</h2><h3>Description</h3><p>Calculate a dendrogram for given input data.</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>cluster</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>ward</b> - ward cluster method</li><li><b>single</b> - single cluster method</li><li><b>complete</b> - complete cluster method</li><li><b>mcquitty</b> - mcquitty cluster method</li><li><b>median</b> - median cluster method</li><li><b>centroid</b> - centroid cluster method</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>columns</b> (list of string)</li><p>column id</p><p>This parameter must be passed in the message body.</p><li><b>data</b> (list of list)</li><p>ARRAY(0x20e4f28)</p><p>This parameter must be passed in the message body.</p><li><b>distance</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>bray-curtis</b> - bray-curtis distance method</li><li><b>euclidean</b> - euclidean distance method</li><li><b>maximum</b> - maximum distance method</li><li><b>manhattan</b> - manhattan distance method</li><li><b>canberra</b> - canberra distance method</li><li><b>minkowski</b> - minkowski distance method</li><li><b>difference</b> - difference distance method</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>norm</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>DESeq_blind</b> - DESeq_blind normalization method</li><li><b>standardize</b> - standardize normalization method</li><li><b>quantile</b> - quantile normalization method</li><li><b>DESeq_per_condition</b> - DESeq_per_condition normalization method</li><li><b>DESeq_pooled</b> - DESeq_pooled normalization method</li><li><b>DESeq_pooled_CR</b> - DESeq_pooled_CR normalization method</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>raw</b> (boolean)</p><p>option to use raw data (not normalize)</p><p>This parameter must be passed in the message body.</p><li><b>rows</b> (list of string)</li><p>row id</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"raw":0,"cluster":"mcquitty","columns": ["mgm4441619.3","mgm4441656.4","mgm4441680.3","mgm4441681.3"], "rows": ["Eukaryota","Bacteria","Archaea"], "data": [[135,410,848,1243],[4397,6529,71423,204413],[1422,2156,874,1138]]}' "http://kbase.us/services/communities/1/compute/heatmap"<li>retrieve dendrogram of normalized input abundances using 'mcquitty' cluster method</ul><h3>Return Attributes</h3><ul><li><b>coldend</b> (object)</li><p>dendrogram object for columns</p><li><b>colindex</b> (list of floats)</li><p>column id index</p><li><b>columns</b> (list of strings)</li><p>column id</p><li><b>data</b> (list of lists of floats)</li><p>normalized value</p><li><b>rowdend</b> (object)</li><p>dendrogram object for rows</p><li><b>rowindex</b> (list of floats)</li><p>row id index</p><li><b>rows</b> (list of strings)</li><p>row id</p></p></ul><h2>pcoa - request</h2><h3>Description</h3><p>Calculate a PCoA for given input data.</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>columns</b> (list of string)</li><p>column id</p><p>This parameter must be passed in the message body.</p><li><b>data</b> (list of list)</li><p>ARRAY(0x20e83b0)</p><p>This parameter must be passed in the message body.</p><li><b>distance</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>bray-curtis</b> - bray-curtis distance method</li><li><b>euclidean</b> - euclidean distance method</li><li><b>maximum</b> - maximum distance method</li><li><b>manhattan</b> - manhattan distance method</li><li><b>canberra</b> - canberra distance method</li><li><b>minkowski</b> - minkowski distance method</li><li><b>difference</b> - difference distance method</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>norm</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>DESeq_blind</b> - DESeq_blind normalization method</li><li><b>standardize</b> - standardize normalization method</li><li><b>quantile</b> - quantile normalization method</li><li><b>DESeq_per_condition</b> - DESeq_per_condition normalization method</li><li><b>DESeq_pooled</b> - DESeq_pooled normalization method</li><li><b>DESeq_pooled_CR</b> - DESeq_pooled_CR normalization method</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>raw</b> (boolean)</p><p>option to use raw data (not normalize)</p><p>This parameter must be passed in the message body.</p><li><b>rows</b> (list of string)</li><p>row id</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"raw":1,"distance":"euclidean","columns": ["mgm4441619.3","mgm4441656.4","mgm4441680.3","mgm4441681.3"], "rows": ["Eukaryota","Bacteria","Archaea"], "data": [[135,410,848,1243],[4397,6529,71423,204413],[1422,2156,874,1138]]}' "http://kbase.us/services/communities/1/compute/pcoa"<li>retrieve PCO of raw input abundances using 'euclidean' distance method</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - column id</li><li><b>pco (list of floats)</b> - principal component value</li></ul><br><li><b>pco</b> (list of floats)</li><p>average principal component value</p></p></ul><hr><h1><a name='download' style='padding-top: 50px;'>download</a></h1><h2>Description</h2><p>An analysis file from the processing of a metagenome from a specific stage in its analysis</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>instance - request</h2><h3>Description</h3><p>Returns a single sequence file.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>file</b> (string)</p><p>file name or identifier</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>link</b> (boolean)</p><p>if true return one time link for download and not file stream</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique metagenome identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/download/mgm4447943.3?file=350.1<li>download fasta file of gene-called protein sequences (from stage 350)</ul><h3>Return Attributes</h3><ul><li><b>data</b> (file)</li><p>requested analysis file</p></p></ul><h2>setlist - request</h2><h3>Description</h3><p>Returns a list of sets of sequence files for the given id.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>stage</b> (string)</p><p>stage name or identifier</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique metagenome identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/download/mgm4447943.3?stage=650<li>view all available files from stage 650</ul><h3>Return Attributes</h3><ul><li><b>file_id</b> (string)</li><p>unique identifier of file in stage</p><li><b>file_name</b> (string)</li><p>name of the analysis file</p><li><b>id</b> (string)</li><p>unique metagenome identifier</p><li><b>stage_id</b> (string)</li><p>three digit numerical identifier of the stage</p><li><b>stage_name</b> (string)</li><p>name of the stage in processing of this file</p><li><b>stage_type</b> (string)</li><p>type of the analysis file within a stage, i.e. passed or removed for quality control steps</p><li><b>url</b> (string)</li><p>url for retrieving this analysis file</p></p></ul><hr><h1><a name='heartbeat' style='padding-top: 50px;'>heartbeat</a></h1><h2>Description</h2><p>Status of services</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>instance - request</h2><h3>Description</h3><p>Returns the status of a service.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>service</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>FTP</b> - file server</li><li><b>website</b> - MG-RAST website</li><li><b>SHOCK</b> - object storage</li><li><b>SHOCKDB</b> - object storage mongodb</li><li><b>AWE</b> - worker engine</li><li><b>AWEDB</b> - worker engine mongodb</li><li><b>M5NR</b> - non-redundant sequence database</li><li><b>solr</b> - search engine</li><li><b>postgres</b> - analysis database</li><li><b>mySQL</b> - job database</li></ul><br><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X GET -H "auth: auth_key" "http://kbase.us/services/communities/1/heartbeat/M5NR"<li>status of the M5NR service</ul><h3>Return Attributes</h3><ul><li><b>service</b> (string)</li><p>cv</p><li><b>status</b> (boolean)</li><p>service is up or not</p><li><b>url</b> (url)</li><p>resource location of this resource</p></p></ul><hr><h1><a name='inbox' style='padding-top: 50px;'>inbox</a></h1><h2>Description</h2><p>inbox receives user inbox data upload, requires authentication, see http://blog.metagenomics.anl.gov/mg-rast-v3-2-faq/#api_submission for details</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>view - request</h2><h3>Description</h3><p>lists the contents of the user inbox</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for file</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X GET -H "auth: auth_key" "http://kbase.us/services/communities/1/inbox"<li>lists the contents of the user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>files</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>checksum (string)</b> - md5 checksum of file</li><li><b>filename (string)</b> - path of file from within user inbox</li><li><b>filesize (string)</b> - disk size of file in bytes</li><li><b>timestamp (string)</b> - timestamp of file</li></ul><br><li><b>id</b> (string)</li><p>user id</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>view_pending - request</h2><h3>Description</h3><p>view status of AWE inbox actions</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>completed</b> (boolean)</p><p>If true show the given state</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>deleted</b> (boolean)</p><p>If true show the given state</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>in-progress</b> (boolean)</p><p>If true show the given state</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>pending</b> (boolean)</p><p>If true show the given state</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>queued</b> (boolean)</p><p>If true show the given state</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>suspend</b> (boolean)</p><p>If true show the given state</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X GET -H "auth: auth_key" "http://kbase.us/services/communities/1/inbox/pending?queued&completed"<li>rename file 'sequences.fastq' in user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>upload - request</h2><h3>Description</h3><p>receives user inbox data upload</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>upload</b> (file)</p><p>file to upload to inbox</p><p>This parameter must be passed in the message body.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X POST -H "auth: auth_key" -F "upload=@sequences.fastq" "http://kbase.us/services/communities/1/inbox"<li>upload file 'sequences.fastq' to user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>delete - request</h2><h3>Description</h3><p>delete indicated file from inbox</p><p>This is a synchronous DELETE request.</p><h3>Parameters</h3><ul><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for file</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X DELETE -H "auth: auth_key" "http://kbase.us/services/communities/1/inbox/cfb3d9e1-c9ba-4260-95bf-e410c57b1e49"<li>delete file 'sequences.fastq' from user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>file_info - request</h2><h3>Description</h3><p>get basic file info - returns results and updates shock node</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for file</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X GET -H "auth: auth_key" "http://kbase.us/services/communities/1/inbox/info/cfb3d9e1-c9ba-4260-95bf-e410c57b1e49"<li>get basic info for file with given id in user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>user id</p><li><b>stats_info</b> (hash)</li><p>key value pairs describing file info</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>validate_metadata - request</h2><h3>Description</h3><p>validate metadata spreadsheet in inbox</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for file</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X GET -H "auth: auth_key" "http://kbase.us/services/communities/1/inbox/validate/cfb3d9e1-c9ba-4260-95bf-e410c57b1e49"<li>validate metadata file with given id in user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>seq_stats - request</h2><h3>Description</h3><p>runs sequence stats on file in user inbox - submits AWE job</p><p>This is a asynchronous GET request.</p><h3>Parameters</h3><ul><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for sequence file</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X GET -H "auth: auth_key" "http://kbase.us/services/communities/1/inbox/stats/cfb3d9e1-c9ba-4260-95bf-e410c57b1e49"<li>runs seq stats on file with given id in user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>awe_id</b> (string)</li><p>url/id of awe job</p><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>rename - request</h2><h3>Description</h3><p>rename indicated file from inbox</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>file</b> (string)</p><p>RFC 4122 UUID for file</p><p>This parameter must be passed in the message body.</p><li><b>name</b> (string)</p><p>new name for file</p><p>This parameter must be passed in the message body.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for file</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X POST -H "auth: auth_key" "http://kbase.us/services/communities/1/inbox/rename"<li>rename file in user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>sff_to_fastq - request</h2><h3>Description</h3><p>create fastq file from sff file - submits AWE job</p><p>This is a asynchronous POST request.</p><h3>Parameters</h3><ul><li><b>sff_file</b> (string)</p><p>RFC 4122 UUID for sff file</p><p>This parameter must be passed in the message body.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X POST -H "auth: auth_key" -F "sff_file=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" "http://kbase.us/services/communities/1/inbox/sff2fastq"<li>create fastq file from sff file with given id in user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>awe_id</b> (string)</li><p>url/id of awe job</p><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>demultiplex - request</h2><h3>Description</h3><p>demultiplex seq file with barcode file - submits AWE job</p><p>This is a asynchronous POST request.</p><h3>Parameters</h3><ul><li><b>barcode_file</b> (string)</p><p>RFC 4122 UUID for barcode mapping file</p><p>This parameter must be passed in the message body.</p><li><b>index_file</b> (string)</p><p>RFC 4122 UUID for index file (optional)</p><p>This parameter must be passed in the message body.</p><li><b>rc_index</b> (boolean)</p><p>If true barcodes in index file are reverse compliment of mapping file, default is false</p><p>This parameter must be passed in the message body.</p><li><b>seq_file</b> (string)</p><p>RFC 4122 UUID for sequence file</p><p>This parameter must be passed in the message body.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X POST -H "auth: auth_key" -F "seq_file=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" -F "barcode_file=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" "http://kbase.us/services/communities/1/inbox/demultiplex"<li>demultiplex seq file with barcode file for given ids in user inbox, auth is required</ul><h3>Return Attributes</h3><ul><li><b>awe_id</b> (string)</li><p>url/id of awe job</p><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>pair_join - request</h2><h3>Description</h3><p>merge overlapping paired-end fastq files - submits AWE job</p><p>This is a asynchronous POST request.</p><h3>Parameters</h3><ul><li><b>output</b> (string)</p><p>prefix for output file, default is a random uuid</p><p>This parameter must be passed in the message body.</p><li><b>pair_file_1</b> (string)</p><p>RFC 4122 UUID for pair 1 file</p><p>This parameter must be passed in the message body.</p><li><b>pair_file_2</b> (string)</p><p>RFC 4122 UUID for pair 2 file</p><p>This parameter must be passed in the message body.</p><li><b>retain</b> (boolean)</p><p>If true retain non-overlapping sequences, default is false</p><p>This parameter must be passed in the message body.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X POST -H "auth: auth_key" -F "retain=1" -F "pair_file_1=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" -F "pair_file_2=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" "http://kbase.us/services/communities/1/inbox/pairjoin"<li>merge overlapping paired-end fastq files for given ids, retain non-overlapping pairs</ul><h3>Return Attributes</h3><ul><li><b>awe_id</b> (string)</li><p>url/id of awe job</p><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><h2>pair_join_demultiplex - request</h2><h3>Description</h3><p>merge overlapping paired-end fastq files and demultiplex based on index file - submits AWE job</p><p>This is a asynchronous POST request.</p><h3>Parameters</h3><ul><li><b>barcode_file</b> (string)</p><p>RFC 4122 UUID for barcode mapping file</p><p>This parameter must be passed in the message body.</p><li><b>index_file</b> (string)</p><p>RFC 4122 UUID for index file</p><p>This parameter must be passed in the message body.</p><li><b>pair_file_1</b> (string)</p><p>RFC 4122 UUID for pair 1 file</p><p>This parameter must be passed in the message body.</p><li><b>pair_file_2</b> (string)</p><p>RFC 4122 UUID for pair 2 file</p><p>This parameter must be passed in the message body.</p><li><b>rc_index</b> (boolean)</p><p>If true barcodes in index file are reverse compliment of mapping file, default is false</p><p>This parameter must be passed in the message body.</p><li><b>retain</b> (boolean)</p><p>If true retain non-overlapping sequences, default is false</p><p>This parameter must be passed in the message body.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X POST -H "auth: auth_key" -F "pair_file_1=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" -F "pair_file_2=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" -F "index_file=cfb3d9e1-c9ba-4260-95bf-e410c57b1e49" "http://kbase.us/services/communities/1/inbox/pairjoin_demultiplex"<li>merge overlapping paired-end fastq files then demultiplex with index file, for given ids</ul><h3>Return Attributes</h3><ul><li><b>awe_id</b> (string)</li><p>url/id of awe job</p><li><b>id</b> (string)</li><p>user id</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user login</p></p></ul><hr><h1><a name='library' style='padding-top: 50px;'>library</a></h1><h2>Description</h2><p>A library of metagenomic samples from some environment</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>query - request</h2><h3>Description</h3><p>Returns a set of data matching the query criteria.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>id</b> - return data objects ordered by id</li><li><b>name</b> - return data objects ordered by name</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>verbosity</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>minimal</b> - returns only minimal information</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/library?limit=20&order=name<li>retrieve the first 20 libraries ordered by name</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>created (date)</b> - time the object was first created</li><li><b>id (string)</b> - unique object identifier</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li><li><b>metagenome (reference metagenome)</b> - reference to the related metagenome object</li><li><b>name (string)</b> - human readable identifier</li><li><b>project (reference project)</b> - reference to the project object</li><li><b>sample (reference sample)</b> - reference to the related sample object</li><li><b>sequencesets (list of reference sequencesets)</b> - a list of references to the related sequence sets</li><li><b>url (uri)</b> - resource location of this object instance</li><li><b>version (integer)</b> - version of the object</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>order</b> (string)</li><p>name of the attribute the returned data is ordered by</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p></p></ul><h2>instance - request</h2><h3>Description</h3><p>Returns a single data object.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>verbosity</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>minimal</b> - returns only minimal information</li><li><b>verbose</b> - returns a standard subselection of metadata</li><li><b>full</b> - returns all connected metadata</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique object identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/library/mgl52924?verbosity=full<li>retrieve all data for library mgl52924</ul><h3>Return Attributes</h3><ul><li><b>created</b> (date)</li><p>time the object was first created</p><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>metadata</b> (hash)</li><p>key value pairs describing metadata</p><li><b>metagenome</b> (reference metagenome)</li><p>reference to the related metagenome object</p><li><b>name</b> (string)</li><p>human readable identifier</p><li><b>project</b> (reference project)</li><p>reference to the project object</p><li><b>sample</b> (reference sample)</li><p>reference to the related sample object</p><li><b>sequencesets</b> (list of reference sequencesets)</li><p>a list of references to the related sequence sets</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of the object</p></p></ul><hr><h1><a name='m5nr' style='padding-top: 50px;'>m5nr</a></h1><h2>Description</h2><p>M5NR provides data through a comprehensive non-redundant protein / rRNA database</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>ontology - request</h2><h3>Description</h3><p>Return functional hierarchy</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>exact</b> (boolean)</p><p>if true return only those ontologies that exactly match filter, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter</b> (string)</p><p>text of ontology group (filter_level) to filter by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>function</b> - bottom function ontology level</li><li><b>level3</b> - function ontology level</li><li><b>level2</b> - function ontology level</li><li><b>level1</b> - top function ontology level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>min_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>function</b> - bottom function ontology level</li><li><b>level3</b> - function ontology level</li><li><b>level2</b> - function ontology level</li><li><b>level1</b> - top function ontology level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>Subsystems</b> - ontology database, type ontology only</li><li><b>NOG</b> - ontology database, type ontology only</li><li><b>COG</b> - ontology database, type ontology only</li><li><b>KO</b> - ontology database, type ontology only</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/ontology?source=Subsystems&min_level=level3<li>retrieve subsystems hierarchy for the top 3 levels</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - ontology ID</li><li><b>level1 (string)</b> - ontology top level</li><li><b>level2 (string)</b> - ontology level 2</li><li><b>level3 (string)</b> - ontology level 3</li><li><b>level4 (string)</b> - ontology bottom level</li><li><b>source (string)</b> - source name</li></ul><br><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>taxonomy - request</h2><h3>Description</h3><p>Return organism hierarchy</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>exact</b> (boolean)</p><p>if true return only those taxonomies that exactly match filter, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter</b> (string)</p><p>text of taxonomy group (filter_level) to filter by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>min_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/taxonomy?filter=Bacteroidetes&filter_level=phylum&min_level=genus<li>retrieve all class level taxa that belong to Bacteroidetes</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>class (string)</b> - organism class</li><li><b>domain (string)</b> - organism domain</li><li><b>family (string)</b> - organism family</li><li><b>genus (string)</b> - organism genus</li><li><b>ncbi_tax_id (int)</b> - organism ncbi id</li><li><b>order (string)</b> - organism order</li><li><b>organism (string)</b> - organism name</li><li><b>phylum (string)</b> - organism phylum</li><li><b>species (string)</b> - organism species</li></ul><br><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>sources - request</h2><h3>Description</h3><p>Return all sources in M5NR</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/sources<li>retrieve all data sources for M5NR</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>source object</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>accession - request</h2><h3>Description</h3><p>Return annotation of given source protein ID</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique identifier from source DB</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/accession/YP_003268079.1<li>retrieve M5NR data for accession ID 'YP_003268079.1'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>alias - request</h2><h3>Description</h3><p>Return annotations for alias IDs containing the given text</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>exact</b> (boolean)</p><p>if true return only those annotations that exactly match input text, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>text</b> (string)</p><p>text string of partial alias</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/alias/IPR001478<li>retrieve M5NR data for db_xref ID 'IPR001478'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>md5 - request</h2><h3>Description</h3><p>Return annotation(s) or sequence of given md5sum (M5NR ID)</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>format</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>fasta</b> - return sequences in fasta format</li><li><b>json</b> - return sequences in json struct</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>sequence</b> (boolean)</p><p>if true return sequence output, else return annotation output, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique identifier in form of md5 checksum</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/md5/000821a2e2f63df1a3873e4b280002a8?source=InterPro<li>retrieve InterPro M5NR data for md5sum '000821a2e2f63df1a3873e4b280002a8'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>function - request</h2><h3>Description</h3><p>Return annotations for function names containing the given text</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>exact</b> (boolean)</p><p>if true return only those annotations that exactly match input text, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>inverse</b> (boolean)</p><p>if true return only those annotations that do not match input text, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>text</b> (string)</p><p>text string of partial function name</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/function/sulfatase?source=GenBank<li>retrieve GenBank M5NR data for function names containing string 'sulfatase'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>organism - request</h2><h3>Description</h3><p>Return annotations for organism names containing the given text</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>exact</b> (boolean)</p><p>if true return only those annotations that exactly match input text, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>inverse</b> (boolean)</p><p>if true return only those annotations that do not match input text, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>tax_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>strain</b> - bottom organism taxonomic level</li><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>text</b> (string)</p><p>text string of partial organism name</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/organism/akkermansia?source=KEGG<li>retrieve KEGG M5NR data for organism names containing string 'akkermansia'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>sequence - request</h2><h3>Description</h3><p>Return annotation(s) for md5sum (M5NR ID) of given sequence</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>text</b> (string)</p><p>text string of protein sequence</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/m5nr/sequence/MAGENHQWQGSIL?source=TrEMBL<li>retrieve TrEMBL M5NR data for md5sum of sequence 'MAGENHQWQGSIL'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>accession - request</h2><h3>Description</h3><p>Return annotations of given source protein IDs</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>data</b> (list of string)</li><p>unique identifier from source DB</p><p>This parameter must be passed in the message body.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This parameter must be passed in the message body.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This parameter must be passed in the message body.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This parameter must be passed in the message body.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"order":"function","data":["YP_003268079.1","COG1764"]}' "http://kbase.us/services/communities/1/m5nr/accession"<li>retrieve M5NR data for accession IDs 'YP_003268079.1' and 'COG1764' ordered by function</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>alias - request</h2><h3>Description</h3><p>Return annotations for aliases containing the given texts</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>data</b> (list of string)</li><p>text string of partial alias ID</p><p>This parameter must be passed in the message body.</p><li><b>exact</b> (boolean)</p><p>if true return only those annotations that exactly match input text, default is false</p><p>This parameter must be passed in the message body.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This parameter must be passed in the message body.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This parameter must be passed in the message body.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This parameter must be passed in the message body.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This parameter must be passed in the message body.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"order":"function","data":["IPR001478","TIGR02124"]}' "http://kbase.us/services/communities/1/m5nr/alias"<li>retrieve M5NR data for aliases containing string 'IPR001478' and 'TIGR02124'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>md5 - request</h2><h3>Description</h3><p>Return annotations or sequences of given md5sums (M5NR ID)</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>data</b> (list of string)</li><p>unique identifier in form of md5 checksum</p><p>This parameter must be passed in the message body.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This parameter must be passed in the message body.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This parameter must be passed in the message body.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This parameter must be passed in the message body.</p><li><b>sequence</b> (boolean)</p><p>if true return sequence output, else return annotation output, default is false</p><p>This parameter must be passed in the message body.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This parameter must be passed in the message body.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"source":"InterPro","data":["000821a2e2f63df1a3873e4b280002a8","15bf1950bd9867099e72ea6516e3d602"]}' "http://kbase.us/services/communities/1/m5nr/md5"<li>retrieve InterPro M5NR data for md5s '000821a2e2f63df1a3873e4b280002a8' and '15bf1950bd9867099e72ea6516e3d602'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>function - request</h2><h3>Description</h3><p>Return annotations for function names containing the given texts</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>data</b> (list of string)</li><p>text string of partial function name</p><p>This parameter must be passed in the message body.</p><li><b>exact</b> (boolean)</p><p>if true return only those annotations that exactly match input text, default is false</p><p>This parameter must be passed in the message body.</p><li><b>inverse</b> (boolean)</p><p>if true return only those annotations that do not match input text, default is false</p><p>This parameter must be passed in the message body.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This parameter must be passed in the message body.</p><li><b>md5s</b> (list of string)</li><p>md5 to constrain search by</p><p>This parameter must be passed in the message body.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This parameter must be passed in the message body.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This parameter must be passed in the message body.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This parameter must be passed in the message body.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"source":"GenBank","limit":50,"data":["sulfatase","phosphatase"]}' "http://kbase.us/services/communities/1/m5nr/function"<li>retrieve top 50 GenBank M5NR data for function names containing string 'sulfatase' or 'phosphatase'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>organism - request</h2><h3>Description</h3><p>Return annotations for organism names containing the given texts</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>data</b> (list of string)</li><p>text string of partial organism name</p><p>This parameter must be passed in the message body.</p><li><b>exact</b> (boolean)</p><p>if true return only those annotations that exactly match input text, default is false</p><p>This parameter must be passed in the message body.</p><li><b>inverse</b> (boolean)</p><p>if true return only those annotations that do not match input text, default is false</p><p>This parameter must be passed in the message body.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This parameter must be passed in the message body.</p><li><b>md5s</b> (list of string)</li><p>md5 to constrain search by</p><p>This parameter must be passed in the message body.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This parameter must be passed in the message body.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This parameter must be passed in the message body.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This parameter must be passed in the message body.</p><li><b>tax_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>strain</b> - bottom organism taxonomic level</li><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This parameter must be passed in the message body.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"source":"KEGG","order":"accession","data":["akkermansia","yersinia"]}' "http://kbase.us/services/communities/1/m5nr/organism"<li>retrieve KEGG M5NR data (ordered by accession ID) for organism names containing string 'akkermansia' or 'yersinia'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><h2>sequence - request</h2><h3>Description</h3><p>Return annotations for md5s (M5NR ID) of given sequences</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>data</b> (list of string)</li><p>text string of protein sequence</p><p>This parameter must be passed in the message body.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This parameter must be passed in the message body.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This parameter must be passed in the message body.</p><li><b>order</b> (string)</p><p>name of the attribute the returned data is ordered by</p><p>This parameter must be passed in the message body.</p><li><b>source</b> (string)</p><p>source name to restrict search by</p><p>This parameter must be passed in the message body.</p><li><b>version</b> (integer)</p><p>M5NR version, default 10</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -d '{"source":"KEGG","order":"source","data":["MAGENHQWQGSIL","MAGENHQWQGSIL"]}' "http://kbase.us/services/communities/1/m5nr/sequence"<li>retrieve M5NR data ordered by source for sequences 'MAGENHQWQGSIL' and 'MAGENHQWQGSIL'</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>accession (string)</b> - unique identifier given by source</li><li><b>alias (list of strings)</b> - db_xref aliases</li><li><b>function (string)</b> - function annotation</li><li><b>md5 (string)</b> - md5 checksum - M5NR ID</li><li><b>ncbi_tax_id (int)</b> - organism ncbi tax_id</li><li><b>organism (string)</b> - organism annotation</li><li><b>source (string)</b> - source name</li><li><b>type (string)</b> - source type</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of M5NR</p></p></ul><hr><h1><a name='matrix' style='padding-top: 50px;'>matrix</a></h1><h2>Description</h2><p>A profile in biom format that contains abundance counts</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>organism - request</h2><h3>Description</h3><p>Returns a BIOM object.</p><p>This is a synchronous or asynchronous GET request.</p><h3>Parameters</h3><ul><li><b>asynchronous</b> (boolean)</p><p>if true return process id to query status resource for results, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>evalue</b> (int)</p><p>negative exponent value for maximum e-value cutoff: default is 5</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter</b> (string)</p><p>filter the return results to only include abundances based on genes with this function</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>function</b> - bottom function ontology level</li><li><b>level3</b> - function ontology level</li><li><b>level2</b> - function ontology level</li><li><b>level1</b> - top function ontology level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>Subsystems</b> - ontology database, type ontology only</li><li><b>NOG</b> - ontology database, type ontology only</li><li><b>COG</b> - ontology database, type ontology only</li><li><b>KO</b> - ontology database, type ontology only</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>grep</b> (string)</p><p>filter the return results to only include annotations that contain this text</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>group_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>strain</b> - bottom organism taxonomic level</li><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>hide_metadata</b> (boolean)</p><p>if true do not return metagenome metadata in 'columns' object, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>hit_type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>all</b> - returns results based on all organisms that map to top hit per read-feature</li><li><b>single</b> - returns results based on a single organism for top hit per read-feature</li><li><b>lca</b> - returns results based on the Least Common Ancestor for all organisms (M5NR+M5RNA only) that map to hits from a read-feature</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>one or more metagenome or project unique identifier</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>identity</b> (int)</p><p>percent value for minimum % identity cutoff: default is 60</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>length</b> (int)</p><p>value for minimum alignment length cutoff: default is 15</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>result_type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>abundance</b> - number of reads with hits in annotation</li><li><b>evalue</b> - average e-value exponent of hits in annotation</li><li><b>identity</b> - average percent identity of hits in annotation</li><li><b>length</b> - average alignment length of hits in annotation</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>M5NR</b> - comprehensive protein database</li><li><b>M5RNA</b> - comprehensive RNA database</li><li><b>RefSeq</b> - protein database, type organism, function, feature</li><li><b>GenBank</b> - protein database, type organism, function, feature</li><li><b>IMG</b> - protein database, type organism, function, feature</li><li><b>SEED</b> - protein database, type organism, function, feature</li><li><b>TrEMBL</b> - protein database, type organism, function, feature</li><li><b>SwissProt</b> - protein database, type organism, function, feature</li><li><b>PATRIC</b> - protein database, type organism, function, feature</li><li><b>KEGG</b> - protein database, type organism, function, feature</li><li><b>RDP</b> - RNA database, type organism, function, feature</li><li><b>Greengenes</b> - RNA database, type organism, function, feature</li><li><b>LSU</b> - RNA database, type organism, function, feature</li><li><b>SSU</b> - RNA database, type organism, function, feature</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>taxid</b> (boolean)</p><p>if true, return annotation ID as NCBI tax id. Only for group_levels with a tax_id</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (int)</p><p>M5NR version, default 1</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/matrix/organism?id=mgm4447943.3&id=mgm4447192.3&id=mgm4447102.3&id=mgm4447103.3&group_level=family&source=RefSeq&result_type=abundance&evalue=15<li>retrieve abundance matrix of RefSeq organism annotations at family taxa for listed metagenomes at evalue < e-15</ul><h3>Return Attributes</h3><ul><li><b>columns</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique metagenome identifier</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li></ul><br><li><b>data</b> (list of lists of floats)</li><p>the matrix values</p><li><b>date</b> (date)</li><p>time the output data was generated</p><li><b>format</b> (string)</li><p>format specification name</p><li><b>format_url</b> (string)</li><p>url to the format specification</p><li><b>generated_by</b> (string)</li><p>identifier of the data generator</p><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>matrix_element_type</b> (string)</li><p>data type of the elements in the return matrix</p><li><b>matrix_element_value</b> (string)</li><p>result_type of the elements in the return matrix</p><li><b>matrix_type</b> (string)</li><p>type of the data encoding matrix (dense or sparse)</p><li><b>rows</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique annotation text</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li></ul><br><li><b>shape</b> (list of integers)</li><p>list of the dimension sizes of the return matrix</p><li><b>type</b> (string)</li><p>type of the data in the return table (taxon, function or gene)</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p></p></ul><h2>function - request</h2><h3>Description</h3><p>Returns a BIOM object.</p><p>This is a synchronous or asynchronous GET request.</p><h3>Parameters</h3><ul><li><b>asynchronous</b> (boolean)</p><p>if true return process id to query status resource for results, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>evalue</b> (int)</p><p>negative exponent value for maximum e-value cutoff: default is 5</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter</b> (string)</p><p>filter the return results to only include abundances based on genes with this organism</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>strain</b> - bottom organism taxonomic level</li><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>M5NR</b> - comprehensive protein database</li><li><b>M5RNA</b> - comprehensive RNA database</li><li><b>RefSeq</b> - protein database, type organism, function, feature</li><li><b>GenBank</b> - protein database, type organism, function, feature</li><li><b>IMG</b> - protein database, type organism, function, feature</li><li><b>SEED</b> - protein database, type organism, function, feature</li><li><b>TrEMBL</b> - protein database, type organism, function, feature</li><li><b>SwissProt</b> - protein database, type organism, function, feature</li><li><b>PATRIC</b> - protein database, type organism, function, feature</li><li><b>KEGG</b> - protein database, type organism, function, feature</li><li><b>RDP</b> - RNA database, type organism, function, feature</li><li><b>Greengenes</b> - RNA database, type organism, function, feature</li><li><b>LSU</b> - RNA database, type organism, function, feature</li><li><b>SSU</b> - RNA database, type organism, function, feature</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>grep</b> (string)</p><p>filter the return results to only include annotations that contain this text</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>group_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>function</b> - bottom function ontology level</li><li><b>level3</b> - function ontology level</li><li><b>level2</b> - function ontology level</li><li><b>level1</b> - top function ontology level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>hide_metadata</b> (boolean)</p><p>if true do not return metagenome metadata in 'columns' object, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>one or more metagenome or project unique identifier</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>identity</b> (int)</p><p>percent value for minimum % identity cutoff: default is 60</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>length</b> (int)</p><p>value for minimum alignment length cutoff: default is 15</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>result_type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>abundance</b> - number of reads with hits in annotation</li><li><b>evalue</b> - average e-value exponent of hits in annotation</li><li><b>identity</b> - average percent identity of hits in annotation</li><li><b>length</b> - average alignment length of hits in annotation</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>Subsystems</b> - ontology database, type ontology only</li><li><b>NOG</b> - ontology database, type ontology only</li><li><b>COG</b> - ontology database, type ontology only</li><li><b>KO</b> - ontology database, type ontology only</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (int)</p><p>M5NR version, default 1</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/matrix/function?id=mgm4447943.3&id=mgm4447192.3&id=mgm4447102.3&id=mgm4447103.3&group_level=level3&source=Subsystems&result_type=abundance&identity=80&filter_level=phylum&filter=Firmicutes<li>retrieve abundance matrix of Subsystem annotations at level3 for listed metagenomes at % identity > 80, filtered to return annotations only in phylum Firmicutes</ul><h3>Return Attributes</h3><ul><li><b>columns</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique metagenome identifier</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li></ul><br><li><b>data</b> (list of lists of floats)</li><p>the matrix values</p><li><b>date</b> (date)</li><p>time the output data was generated</p><li><b>format</b> (string)</li><p>format specification name</p><li><b>format_url</b> (string)</li><p>url to the format specification</p><li><b>generated_by</b> (string)</li><p>identifier of the data generator</p><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>matrix_element_type</b> (string)</li><p>data type of the elements in the return matrix</p><li><b>matrix_element_value</b> (string)</li><p>result_type of the elements in the return matrix</p><li><b>matrix_type</b> (string)</li><p>type of the data encoding matrix (dense or sparse)</p><li><b>rows</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique annotation text</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li></ul><br><li><b>shape</b> (list of integers)</li><p>list of the dimension sizes of the return matrix</p><li><b>type</b> (string)</li><p>type of the data in the return table (taxon, function or gene)</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p></p></ul><h2>feature - request</h2><h3>Description</h3><p>Returns a BIOM object.</p><p>This is a synchronous or asynchronous GET request.</p><h3>Parameters</h3><ul><li><b>asynchronous</b> (boolean)</p><p>if true return process id to query status resource for results, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>evalue</b> (int)</p><p>negative exponent value for maximum e-value cutoff: default is 5</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter</b> (string)</p><p>filter the return results to only include abundances based on genes with this organism</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>filter_level</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>strain</b> - bottom organism taxonomic level</li><li><b>species</b> - organism type level</li><li><b>genus</b> - organism taxonomic level</li><li><b>family</b> - organism taxonomic level</li><li><b>order</b> - organism taxonomic level</li><li><b>class</b> - organism taxonomic level</li><li><b>phylum</b> - organism taxonomic level</li><li><b>domain</b> - top organism taxonomic level</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>hide_annotation</b> (boolean)</p><p>if true do not return feature metadata in 'rows' object, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>hide_metadata</b> (boolean)</p><p>if true do not return metagenome metadata in 'columns' object, default is false</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>one or more metagenome or project unique identifier</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>identity</b> (int)</p><p>percent value for minimum % identity cutoff: default is 60</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>length</b> (int)</p><p>value for minimum alignment length cutoff: default is 15</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>result_type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>abundance</b> - number of reads with hits in annotation</li><li><b>evalue</b> - average e-value exponent of hits in annotation</li><li><b>identity</b> - average percent identity of hits in annotation</li><li><b>length</b> - average alignment length of hits in annotation</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>RefSeq</b> - protein database, type organism, function, feature</li><li><b>GenBank</b> - protein database, type organism, function, feature</li><li><b>IMG</b> - protein database, type organism, function, feature</li><li><b>SEED</b> - protein database, type organism, function, feature</li><li><b>TrEMBL</b> - protein database, type organism, function, feature</li><li><b>SwissProt</b> - protein database, type organism, function, feature</li><li><b>PATRIC</b> - protein database, type organism, function, feature</li><li><b>KEGG</b> - protein database, type organism, function, feature</li><li><b>RDP</b> - RNA database, type organism, function, feature</li><li><b>Greengenes</b> - RNA database, type organism, function, feature</li><li><b>LSU</b> - RNA database, type organism, function, feature</li><li><b>SSU</b> - RNA database, type organism, function, feature</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (int)</p><p>M5NR version, default 1</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/matrix/feature?id=mgm4447943.3&id=mgm4447192.3&id=mgm4447102.3&id=mgm4447103.3&source=KEGG&result_type=evalue&length=25<li>retrieve e-value matrix of KEGG protein annotations for listed metagenomes at alignment length > 25</ul><h3>Return Attributes</h3><ul><li><b>columns</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique metagenome identifier</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li></ul><br><li><b>data</b> (list of lists of floats)</li><p>the matrix values</p><li><b>date</b> (date)</li><p>time the output data was generated</p><li><b>format</b> (string)</li><p>format specification name</p><li><b>format_url</b> (string)</li><p>url to the format specification</p><li><b>generated_by</b> (string)</li><p>identifier of the data generator</p><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>matrix_element_type</b> (string)</li><p>data type of the elements in the return matrix</p><li><b>matrix_element_value</b> (string)</li><p>result_type of the elements in the return matrix</p><li><b>matrix_type</b> (string)</li><p>type of the data encoding matrix (dense or sparse)</p><li><b>rows</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique annotation text</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li></ul><br><li><b>shape</b> (list of integers)</li><p>list of the dimension sizes of the return matrix</p><li><b>type</b> (string)</li><p>type of the data in the return table (taxon, function or gene)</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p></p></ul><hr><h1><a name='metadata' style='padding-top: 50px;'>metadata</a></h1><h2>Description</h2><p>Metagenomic metadata is data providing information about one or more aspects of a set sequences from a sample of some environment</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>template - request</h2><h3>Description</h3><p>Returns static template for metadata object relationships and types</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metadata/template<li>metadata template</ul><h3>Return Attributes</h3><ul><li><b>ep</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - environmental package type</li><li><b>value (hash)</b> - hash of metadata objects by label</li></ul><br><li><b>library</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - library type</li><li><b>value (hash)</b> - hash of metadata objects by label</li></ul><br><li><b>project</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - project type</li><li><b>value (hash)</b> - hash of metadata objects by label</li></ul><br><li><b>sample</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - sample type</li><li><b>value (hash)</b> - hash of metadata objects by label</li></ul><br></p></ul><h2>cv - request</h2><h3>Description</h3><p>Returns static controlled vocabularies used in metadata. By default returns all CVs at latest version. If label and version options used, returns those specific values.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>label</b> (string)</p><p>metadata label</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (string)</p><p>version of CV ontology to use</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metadata/cv?label=country<li>metadata controlled vocabularies</ul><h3>Return Attributes</h3><ul><li><b>ont_info</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - metadata label</li><li><b>value (list of strings)</b> - ontology url and ID</li></ul><br><li><b>ontology</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - metadata label</li><li><b>value (list of lists of strings)</b> - ontology term and ID</li></ul><br><li><b>select</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - metadata label</li><li><b>value (list of strings)</b> - CV term</li></ul><br></p></ul><h2>ontology - request</h2><h3>Description</h3><p>Returns static ontology used in metadata for the given name and version.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>name</b> (string)</p><p>ontology name</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>version</b> (string)</p><p>version of ontology to use</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metadata/ontology?name=biome&version=2013-04-27<li>metadata ontology lookup</ul><h3>Return Attributes</h3><ul><li><b>name</b> (string)</li><p>ontology name</p><li><b>nodes</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - ontology ID</li><li><b>value (hash)</b> - hash of information and relationships for given ontology ID</li></ul><br><li><b>rootNode</b> (string)</li><p>ontology ID of root</p><li><b>showRoot</b> (boolean)</li><p>option to show root when displaying</p><li><b>type</b> (string)</li><p>this type</p><li><b>version</b> (string)</li><p>version of this ontology</p></p></ul><h2>version - request</h2><h3>Description</h3><p>Returns all versions available for given ontology name.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>label</b> (string)</p><p>metadata label</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metadata/version?name=biome<li>metadata version lookup</ul><h3>Return Attributes</h3><ul><li><b>biome</b> (list of strings)</li><p>version number</p><li><b>feature</b> (list of strings)</li><p>version number</p><li><b>material</b> (list of strings)</li><p>version number</p></p></ul><h2>view - request</h2><h3>Description</h3><p>Returns list of unique metadata values for given label</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>label</b> (string)</p><p>valid metadata label</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metadata/view/biome<li>all biome values</ul><h3>Return Attributes</h3><ul><li><b>label</b> (string)</li><p>metadata label</p><li><b>total</b> (int)</li><p>count of unique values</p><li><b>values</b> (list of strings)</li><p>metadata value</p></p></ul><h2>export - request</h2><h3>Description</h3><p>Returns full nested metadata for a project in same format as template, or metadata for a single metagenome.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>id</b> (string)</p><p>unique object identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metadata/export/mgp128<li>all metadata for project mgp128</ul><h3>Return Attributes</h3><ul><li><b>data</b> (hash)</li><p>This attribute has a hash structure:</p><ul style='list-style: none;'><li><b>key (string)</b> - metadata label</li><li><b>value (object)</b> - project metadata objects</li></ul><br><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>name</b> (string)</li><p>human readable identifier</p><li><b>sampleNum</b> (int)</li><p>number of samples in project</p><li><b>samples</b> (list of objects)</li><p>sample object containing sample metadata, sample libraries, sample envPackage</p></p></ul><h2>import - request</h2><h3>Description</h3><p>Create project with given metadata spreadsheet and metagenome IDs, either upload or shock node</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>metagenome</b> (string)</p><p>unique metagenome ID</p><p>This parameter must be passed in the message body.</p><li><b>node_id</b> (string)</p><p>shock node ID of metadata spreadsheet</p><p>This parameter must be passed in the message body.</p><li><b>upload</b> (file)</p><p>xlsx or xls format spreadsheet with metadata</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -F "metagenome=mgm12345" -F "metagenome=mgm67890" -F "upload=@metadata.xlsx" "http://kbase.us/services/communities/1/metadata/import"<li>create project with metadata from file 'metadata.xlsx'</ul><h3>Return Attributes</h3><ul><li><b>added</b> (list of strings)</li><p>ID of metagenome with metadata added</p><li><b>errors</b> (list of strings)</li><p>error message that may have occurred</p><li><b>project</b> (string)</li><p>unique object identifier</p></p></ul><h2>update - request</h2><h3>Description</h3><p>Update project with given metadata spreadsheet and metagenome IDs, either upload or shock node</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>map_by_id</b> (boolean)</p><p>option to map metadata from spreadsheet to metagenomes using ID, default is name</p><p>This parameter must be passed in the message body.</p><li><b>metagenome</b> (string)</p><p>unique metagenome ID</p><p>This parameter must be passed in the message body.</p><li><b>node_id</b> (string)</p><p>shock node ID of metadata spreadsheet</p><p>This parameter must be passed in the message body.</p><li><b>project</b> (string)</p><p>unique project identifier</p><p>This parameter must be passed in the message body.</p><li><b>upload</b> (file)</p><p>xlsx or xls format spreadsheet with metadata</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -F "project=mgp123" -F "upload=@metadata.xlsx" "http://kbase.us/services/communities/1/metadata/update"<li>update project mgp123 with metadata from file 'metadata.xlsx'</ul><h3>Return Attributes</h3><ul><li><b>added</b> (list of strings)</li><p>ID of metagenome with metadata added</p><li><b>errors</b> (list of strings)</li><p>error message that may have occurred</p><li><b>project</b> (string)</li><p>unique object identifier</p></p></ul><h2>validate - request</h2><h3>Description</h3><p>Validate given metadata spreadsheet, either upload or shock node</p><p>This is a synchronous POST request.</p><h3>Parameters</h3><ul><li><b>node_id</b> (string)</p><p>shock node ID of metadata spreadsheet</p><p>This parameter must be passed in the message body.</p><li><b>upload</b> (file)</p><p>xlsx or xls format spreadsheet with metadata</p><p>This parameter must be passed in the message body.</p></ul><h3>Example</h3><ul><li>curl -X POST -F "upload=@metadata.xlsx" "http://kbase.us/services/communities/1/metadata/validate"<li>validate file 'metadata.xlsx' against MG-RAST metadata template</ul><h3>Return Attributes</h3><ul><li><b>is_valid</b> (boolean)</li><p>the metadata sheet is valid</p><li><b>message</b> (string)</li><p>if not valid, reason why</p><li><b>metadata</b> (object)</li><p>valid metadata object for project and its samples and libraries</p></p></ul><h2>validate - request</h2><h3>Description</h3><p>Validate given metadata value</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>category</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>project</b> - label belongs to project metadata</li><li><b>sample</b> - label belongs to sample metadata</li><li><b>library</b> - label belongs to library metadata</li><li><b>env_package</b> - label belongs to env_package metadata</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>group</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>mixs</b> - label is part of MIxS (minimal) metadata</li><li><b>mims</b> - label is part of MIMS (metagenome) metadata</li><li><b>migs</b> - label is part of MIGS (genome) metadata</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>label</b> (string)</p><p>metadata label</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>value</b> (string)</p><p>metadata value</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>version</b> (string)</p><p>version of CV ontology to use</p><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metadata/validate?category=sample&label=material&value=soil<li>check if 'soil' is a valid term for sample material</ul><h3>Return Attributes</h3><ul><li><b>is_valid</b> (boolean)</li><p>the inputed value is valid for the given category and label</p><li><b>message</b> (string)</li><p>if not valid, reason why</p></p></ul><hr><h1><a name='metagenome' style='padding-top: 50px;'>metagenome</a></h1><h2>Description</h2><p>A metagenome is an analyzed set sequences from a sample of some environment</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>query - request</h2><h3>Description</h3><p>Returns a set of data matching the query criteria.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>direction</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>asc</b> - sort by ascending order</li><li><b>desc</b> - sort by descending order</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>function</b> (string)</p><p>search parameter: query string for function</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>match</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>all</b> - return metagenomes that match all search parameters</li><li><b>any</b> - return metagenomes that match any search parameters</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>md5</b> (string)</p><p>search parameter: md5 checksum of feature sequence</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>metadata</b> (string)</p><p>search parameter: query string for any metadata field</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (string)</p><p>metagenome object field to sort by (default is id)</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>organism</b> (string)</p><p>search parameter: query string for organism</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>status</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>both</b> - returns all data (public and private) user has access to view</li><li><b>public</b> - returns all public data</li><li><b>private</b> - returns private data user has access to view</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>verbosity</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>minimal</b> - returns only minimal information</li><li><b>mixs</b> - returns all GSC MIxS metadata</li><li><b>metadata</b> - returns minimal with metadata</li><li><b>stats</b> - returns minimal with statistics</li><li><b>full</b> - returns all metadata and statistics</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metagenome?limit=20&order=name<li>retrieve the first 20 metagenomes ordered by name</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>PI_firstname (string)</b> - principal investigator's first name</li><li><b>PI_lastname (string)</b> - principal investigator's last name</li><li><b>biome (string)</b> - environmental biome, EnvO term</li><li><b>collection_date (string)</b> - date sample collected</li><li><b>country (string)</b> - country where sample taken</li><li><b>created (date)</b> - time the metagenome was first created</li><li><b>env_package_type (string)</b> - environmental package of sample, GSC term</li><li><b>feature (string)</b> - environmental feature, EnvO term</li><li><b>id (string)</b> - unique metagenome identifier</li><li><b>latitude (string)</b> - latitude where sample taken</li><li><b>location (string)</b> - location where sample taken</li><li><b>longitude (string)</b> - longitude where sample taken</li><li><b>material (string)</b> - environmental material, EnvO term</li><li><b>name (string)</b> - name of metagenome</li><li><b>project_id (string)</b> - id of project containing metagenome</li><li><b>project_name (string)</b> - name of project containing metagenome</li><li><b>seq_method (string)</b> - sequencing method</li><li><b>sequence_type (string)</b> - sequencing type</li><li><b>status (cv of ARRAY(0x2152398)s of privates)</b> - metagenome is private</li><li><b>url (uri)</b> - resource location of this object instance</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>order</b> (string)</li><p>name of the attribute the returned data is ordered by</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of the object</p></p></ul><h2>instance - request</h2><h3>Description</h3><p>Returns a single data object.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>verbosity</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>minimal</b> - returns only minimal information</li><li><b>metadata</b> - returns minimal with metadata</li><li><b>stats</b> - returns minimal with statistics</li><li><b>full</b> - returns all metadata and statistics</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique object identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/metagenome/mgm4447943.3?verbosity=metadata<li>retrieve all metadata for metagenome mgm4447943.3</ul><h3>Return Attributes</h3><ul><li><b>created</b> (date)</li><p>time the metagenome was first created</p><li><b>id</b> (string)</li><p>unique metagenome identifier</p><li><b>library</b> (reference library)</li><p>reference to the related library object</p><li><b>metadata</b> (hash)</li><p>key value pairs describing all metadata</p><li><b>mixs</b> (hash)</li><p>key value pairs describing MIxS metadata</p><li><b>name</b> (string)</li><p>name of metagenome</p><li><b>pipeline_parameters</b> (hash)</li><p>key value pairs describing pipeline parameters</p><li><b>project</b> (reference project)</li><p>reference to the project object</p><li><b>sample</b> (reference sample)</li><p>reference to the related sample object</p><li><b>sequence_type</b> (string)</li><p>sequencing type</p><li><b>statistics</b> (hash)</li><p>key value pairs describing statistics</p><li><b>status</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>public</b> - metagenome is public</li><li><b>private</b> - metagenome is private</li></ul><br><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of the metagenome</p></p></ul><hr><h1><a name='profile' style='padding-top: 50px;'>profile</a></h1><h2>Description</h2><p>A profile in biom format that contains abundance and similarity values</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>instance - request</h2><h3>Description</h3><p>Returns a single data object in BIOM format</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>evalue</b> (int)</p><p>negative exponent value for maximum e-value cutoff: default is 5</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>hit_type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>all</b> - returns results based on all organisms that map to top hit per read-feature</li><li><b>single</b> - returns results based on a single organism for top hit per read-feature</li><li><b>lca</b> - returns results based on the Least Common Ancestor for all organisms (M5NR+M5RNA only) that map to hits from a read-feature</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>identity</b> (int)</p><p>percent value for minimum % identity cutoff: default is 60</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>length</b> (int)</p><p>value for minimum alignment length cutoff: default is 15</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>nocutoff</b> (boolean)</p><p>if true, get data using no cutoffs</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>source</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>M5NR</b> - comprehensive protein database</li><li><b>RefSeq</b> - protein database, type organism, function, feature</li><li><b>GenBank</b> - protein database, type organism, function, feature</li><li><b>IMG</b> - protein database, type organism, function, feature</li><li><b>SEED</b> - protein database, type organism, function, feature</li><li><b>TrEMBL</b> - protein database, type organism, function, feature</li><li><b>SwissProt</b> - protein database, type organism, function, feature</li><li><b>PATRIC</b> - protein database, type organism, function, feature</li><li><b>KEGG</b> - protein database, type organism, function, feature</li><li><b>M5RNA</b> - comprehensive RNA database</li><li><b>RDP</b> - RNA database, type organism, function, feature</li><li><b>Greengenes</b> - RNA database, type organism, function, feature</li><li><b>LSU</b> - RNA database, type organism, function, feature</li><li><b>SSU</b> - RNA database, type organism, function, feature</li><li><b>Subsystems</b> - ontology database, type ontology only</li><li><b>NOG</b> - ontology database, type ontology only</li><li><b>COG</b> - ontology database, type ontology only</li><li><b>KO</b> - ontology database, type ontology only</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>type</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>organism</b> - return organism data</li><li><b>function</b> - return functional data</li><li><b>feature</b> - return feature data</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique object identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Return Attributes</h3><ul><li><b>columns</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique identifier</li><li><b>metadata (hash)</b> - list of metadata, contains the metagenome</li></ul><br><li><b>data</b> (list of lists of floats)</li><p>the matrix values</p><li><b>date</b> (date)</li><p>time the output data was generated</p><li><b>format</b> (string)</li><p>format specification name</p><li><b>format_url</b> (string)</li><p>url to the format specification</p><li><b>generated_by</b> (string)</li><p>identifier of the data generator</p><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>matrix_element_type</b> (string)</li><p>data type of the elements in the return matrix</p><li><b>matrix_type</b> (string)</li><p>type of the data encoding matrix (dense or sparse)</p><li><b>rows</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - unique identifier</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li></ul><br><li><b>shape</b> (list of integers)</li><p>list of the dimension sizes of the return matrix</p><li><b>type</b> (string)</li><p>type of the data in the return table (taxon, function or gene)</p></p></ul><hr><h1><a name='project' style='padding-top: 50px;'>project</a></h1><h2>Description</h2><p>A project is a composition of samples, libraries and metagenomes being analyzed in a global context.</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>query - request</h2><h3>Description</h3><p>Returns a set of data matching the query criteria.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>id</b> - return data objects ordered by id</li><li><b>name</b> - return data objects ordered by name</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>verbosity</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>minimal</b> - returns only minimal information</li><li><b>verbose</b> - returns all metadata</li><li><b>full</b> - returns all metadata and references</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/project?limit=20&order=name<li>retrieve the first 20 projects ordered by name</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>created (date)</b> - time the object was first created</li><li><b>description (string)</b> - a short, comprehensive description of the project</li><li><b>funding_source (string)</b> - the official name of the source of funding of this project</li><li><b>id (string)</b> - unique object identifier</li><li><b>libraries (list of reference librarys)</b> - a list of references to the related library objects</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li><li><b>metagenomes (list of reference metagenomes)</b> - a list of references to the related metagenome objects</li><li><b>name (string)</b> - human readable identifier</li><li><b>pi (string)</b> - the first and last name of the principal investigator of the project</li><li><b>samples (list of reference samples)</b> - a list of references to the related sample objects</li><li><b>status (cv of ARRAY(0x2159fd0)s of privates)</b> - object is private</li><li><b>url (uri)</b> - resource location of this object instance</li><li><b>version (integer)</b> - version of the object</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>order</b> (string)</li><p>name of the attribute the returned data is ordered by</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p></p></ul><h2>instance - request</h2><h3>Description</h3><p>Returns a single data object.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>verbosity</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>minimal</b> - returns only minimal information</li><li><b>verbose</b> - returns all metadata</li><li><b>full</b> - returns all metadata and references</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique object identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/project/mgp128?verbosity=full<li>retrieve all data for project mgp128</ul><h3>Return Attributes</h3><ul><li><b>created</b> (date)</li><p>time the object was first created</p><li><b>description</b> (string)</li><p>a short, comprehensive description of the project</p><li><b>funding_source</b> (string)</li><p>the official name of the source of funding of this project</p><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>libraries</b> (list of reference librarys)</li><p>a list of references to the related library objects</p><li><b>metadata</b> (hash)</li><p>key value pairs describing metadata</p><li><b>metagenomes</b> (list of reference metagenomes)</li><p>a list of references to the related metagenome objects</p><li><b>name</b> (string)</li><p>human readable identifier</p><li><b>pi</b> (string)</li><p>the first and last name of the principal investigator of the project</p><li><b>samples</b> (list of reference samples)</li><p>a list of references to the related sample objects</p><li><b>status</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>public</b> - object is public</li><li><b>private</b> - object is private</li></ul><br><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of the object</p></p></ul><hr><h1><a name='sample' style='padding-top: 50px;'>sample</a></h1><h2>Description</h2><p>A metagenomic sample from some environment.</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>query - request</h2><h3>Description</h3><p>Returns a set of data matching the query criteria.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>limit</b> (integer)</p><p>maximum number of items requested</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>offset</b> (integer)</p><p>zero based index of the first data object to be returned</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>order</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>id</b> - return data objects ordered by id</li><li><b>name</b> - return data objects ordered by name</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/sample?limit=20&order=name<li>retrieve the first 20 samples ordered by name</ul><h3>Return Attributes</h3><ul><li><b>data</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>created (date)</b> - time the object was first created</li><li><b>env_package (object of HASH(0x21602a8)s)</b> - environmental package object</li><li><b>id (string)</b> - unique object identifier</li><li><b>libraries (list of reference librarys)</b> - a list of references to the related library objects</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li><li><b>metagenomes (list of reference metagenomes)</b> - a list of references to the related metagenome objects</li><li><b>name (string)</b> - human readable identifier</li><li><b>project (reference project)</b> - reference to the project of this sample</li><li><b>url (uri)</b> - resource location of this object instance</li><li><b>version (integer)</b> - version of the object</li></ul><br><li><b>limit</b> (integer)</li><p>maximum number of data items returned, default is 10</p><li><b>next</b> (uri)</li><p>link to the previous set or null if this is the first set</p><li><b>offset</b> (integer)</li><p>zero based index of the first returned data item</p><li><b>order</b> (string)</li><p>name of the attribute the returned data is ordered by</p><li><b>prev</b> (uri)</li><p>link to the next set or null if this is the last set</p><li><b>total_count</b> (integer)</li><p>total number of available data items</p></p></ul><h2>instance - request</h2><h3>Description</h3><p>Returns a single data object.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>verbosity</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>minimal</b> - returns only minimal information</li><li><b>verbose</b> - returns all metadata</li><li><b>full</b> - returns all metadata and references</li></ul><br><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>unique object identifier</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/sample/mgs25823?verbosity=full<li>retrieve all data for sample mgs25823</ul><h3>Return Attributes</h3><ul><li><b>created</b> (date)</li><p>time the object was first created</p><li><b>env_package</b> (object)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>created (date)</b> - creation date</li><li><b>id (string)</b> - unique package identifier</li><li><b>metadata (hash)</b> - key value pairs describing metadata</li><li><b>name (string)</b> - name of the package</li><li><b>type (string)</b> - package type</li></ul><br><li><b>id</b> (string)</li><p>unique object identifier</p><li><b>libraries</b> (list of reference librarys)</li><p>a list of references to the related library objects</p><li><b>metadata</b> (hash)</li><p>key value pairs describing metadata</p><li><b>metagenomes</b> (list of reference metagenomes)</li><p>a list of references to the related metagenome objects</p><li><b>name</b> (string)</li><p>human readable identifier</p><li><b>project</b> (reference project)</li><p>reference to the project of this sample</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (integer)</li><p>version of the object</p></p></ul><hr><h1><a name='server' style='padding-top: 50px;'>server</a></h1><h2>Description</h2><p>The server resource returns information about a server.</p><h2>info - request</h2><h3>Description</h3><p>Returns the server information.</p><p>This is a synchronous GET request. This request has no parameters.<h2>instance - request</h2><h3>Description</h3><p>Returns a single user object.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>id</b> (string)</p><p>unique server ID</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>curl -X GET "http://kbase.us/services/communities/1/server/MG-RAST"<li>info for the MG-RAST server</ul><h3>Return Attributes</h3><ul><li><b>basepairs</b> (integer)</li><p>total number of basepairs</p><li><b>id</b> (string)</li><p>unique identifier of this server</p><li><b>info</b> (string)</li><p>informational text, i.e. downtime warnings</p><li><b>metagenomes</b> (integer)</li><p>total number of metagenomes</p><li><b>public_metagenomes</b> (integer)</li><p>total number of public metagenomes</p><li><b>sequences</b> (integer)</li><p>total number of sequences</p><li><b>status</b> (string)</li><p>status of the server</p><li><b>url</b> (uri)</li><p>resource location of this object instance</p><li><b>version</b> (string)</li><p>version number of the server</p></p></ul><hr><h1><a name='submission' style='padding-top: 50px;'>submission</a></h1><h2>Description</h2><p>submission runs input through a series of validation and pre-processing steps, then submits the results to the MG-RAST anaylsis pipeline</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>list - request</h2><h3>Description</h3><p>list all submissions by user</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Return Attributes</h3><ul><li><b>submissions</b> (list of objects)</li><p>This attribute has an object structure:</p><ul style='list-style: none;'><li><b>id (string)</b> - RFC 4122 UUID for submission</li><li><b>status (string)</b> - status of submission</li><li><b>timestamp (string)</b> - timestamp of submission creation</li><li><b>type (string)</b> - type of submission</li></ul><br><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user id</p></p></ul><h2>status - request</h2><h3>Description</h3><p>get status of submission from ID</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>full</b> (boolean)</p><p>if true show full document of running jobs, default is summary</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for submission</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>RFC 4122 UUID for submission</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user id</p></p></ul><h2>delete - request</h2><h3>Description</h3><p>delete all files and running processes for given submission ID</p><p>This is a synchronous DELETE request.</p><h3>Parameters</h3><ul><li><b>full</b> (boolean)</p><p>if true delete all files and metagenomes in mgrast for submission, default just data in inbox</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p><li><b>uuid</b> (string)</p><p>RFC 4122 UUID for submission</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>RFC 4122 UUID for submission</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user id</p></p></ul><h2>submit - request</h2><h3>Description</h3><p>start new submission</p><p>This is a asynchronous POST request.</p><h3>Parameters</h3><ul><li><b>assembled</b> (boolean)</p><p>if true sequences are assembeled, default is false</p><p>This parameter must be passed in the message body.</p><li><b>barcode_file</b> (string)</p><p>RFC 4122 UUID for barcode mapping file</p><p>This parameter must be passed in the message body.</p><li><b>bowtie</b> (boolean)</p><p>if true run bowtie screening, default is true</p><p>This parameter must be passed in the message body.</p><li><b>debug</b> (boolean)</p><p>if true return workflow document instead of submitting</p><p>This parameter must be passed in the message body.</p><li><b>dereplicate</b> (boolean)</p><p>if true run dereplication, default is true</p><p>This parameter must be passed in the message body.</p><li><b>dynamic_trim</b> (boolean)</p><p>if true run qual score dynamic trimmer, default is true</p><p>This parameter must be passed in the message body.</p><li><b>filter_ambig</b> (boolean)</p><p>if true run sequence ambiguous bp filtering, default is true</p><p>This parameter must be passed in the message body.</p><li><b>filter_ln</b> (boolean)</p><p>if true run sequence length filtering, default is true</p><p>This parameter must be passed in the message body.</p><li><b>filter_ln_mult</b> (float)</p><p>sequence length filtering multiplier, default is 2.0</p><p>This parameter must be passed in the message body.</p><li><b>index_file</b> (string)</p><p>RFC 4122 UUID for index (barcode) file</p><p>This parameter must be passed in the message body.</p><li><b>max_ambig</b> (int)</p><p>maximum ambiguous bps to allow through per sequence, default is 5</p><p>This parameter must be passed in the message body.</p><li><b>max_lqb</b> (int)</p><p>maximum number of low-quality bases per read, default is 5</p><p>This parameter must be passed in the message body.</p><li><b>metadata_file</b> (string)</p><p>RFC 4122 UUID for metadata file</p><p>This parameter must be passed in the message body.</p><li><b>mg_name</b> (string)</p><p>name of metagenome for pair-join</p><p>This parameter must be passed in the message body.</p><li><b>min_qual</b> (int)</p><p>quality threshold for low-quality bases, default is 15</p><p>This parameter must be passed in the message body.</p><li><b>multiplex_file</b> (string)</p><p>RFC 4122 UUID for file to demultiplex</p><p>This parameter must be passed in the message body.</p><li><b>pair_file_1</b> (string)</p><p>RFC 4122 UUID for pair 1 file</p><p>This parameter must be passed in the message body.</p><li><b>pair_file_2</b> (string)</p><p>RFC 4122 UUID for pair 2 file</p><p>This parameter must be passed in the message body.</p><li><b>priority</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>never</b></li><li><b>Data will stay private (default)</b></li></ul><br><p>This parameter must be passed in the message body.</p><li><b>project_id</b> (string)</p><p>unique MG-RAST project identifier</p><p>This parameter must be passed in the message body.</p><li><b>project_name</b> (string)</p><p>unique MG-RAST project name</p><p>This parameter must be passed in the message body.</p><li><b>rc_index</b> (boolean)</p><p>If true barcodes in index file are reverse compliment of mapping file, default is false</p><p>This parameter must be passed in the message body.</p><li><b>retain</b> (boolean)</p><p>If true retain non-overlapping sequences, default is false</p><p>This parameter must be passed in the message body.</p><li><b>screen_indexes</b> (controlled vocabulary)</li><p>This parameter value can be chosen from the following (the first being default):</p><ul style='list-style: none;'><li><b>h_sapiens</b></li><li><b>Homo sapiens (default)</b></li></ul><br><p>This parameter must be passed in the message body.</p><li><b>seq_files</b> (list of string)</li><p>RFC 4122 UUID for sequence file</p><p>This parameter must be passed in the message body.</p><li><b>auth</b> (string)</p><p>unique string of text generated by MG-RAST for your account</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Return Attributes</h3><ul><li><b>id</b> (string)</li><p>RFC 4122 UUID for submission</p><li><b>status</b> (string)</li><p>status message</p><li><b>timestamp</b> (string)</li><p>timestamp for return of this query</p><li><b>user</b> (string)</li><p>user id</p></p></ul><hr><h1><a name='validation' style='padding-top: 50px;'>validation</a></h1><h2>Description</h2><p>validates templates for correct structure and data to fit a valid template</p><h2>info - request</h2><h3>Description</h3><p>Returns description of parameters and attributes.</p><p>This is a synchronous GET request. This request has no parameters.<h2>template - request</h2><h3>Description</h3><p>Checks if the referenced JSON structure is a valid template</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>id</b> (string)</p><p>SHOCK template id</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/validation/template/<li>validate the communities metagenomics template</ul><h3>Return Attributes</h3><ul><li><b>error</b> (array of strings)</li><p>array of invalid entries</p><li><b>valid</b> (boolean)</li><p>boolean indicating whether the examined template is valid or not</p></p></ul><h2>data - request</h2><h3>Description</h3><p>Returns a single data object.</p><p>This is a synchronous GET request.</p><h3>Parameters</h3><ul><li><b>template</b> (string)</p><p>SHOCK template id, default is MG-RAST metagenome metadata template</p><p>This is an optional parameter and may be passed in the query string.</p><li><b>id</b> (string)</p><p>SHOCK data id</p><p>This is a required parameter and must be passed as a REST parameter.</p></ul><h3>Example</h3><ul><li>http://kbase.us/services/communities/1/validation/data/?template=<li>validate a JSON data structure against the MG-RAST metagenome metadata template</ul><h3>Return Attributes</h3><ul><li><b>error</b> (array of strings)</li><p>array of invalid entries</p><li><b>valid</b> (boolean)</li><p>boolean indicating whether the examined template is valid or not</p></p></ul><hr>
</ul>
    </div>
  </body>

</html>
